edge-modules/TwinTester/TwinOperator.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TwinTester
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Shared;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public class TwinOperator
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger(nameof(TwinOperator));
+        readonly SemaphoreSlim operationLock = new SemaphoreSlim(1, 1);
+        readonly ReportedPropertyOperation reportedPropertyOperation;
```

Comment:
You can use the base class (or interface if you change to interface) instead of ReportedPropertyOperation  and DesiredPropertyOperation